src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,35 @@
 package mate.academy;
 
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private ConcurrentHashMap events = new ConcurrentHashMap();
+
+    private CopyOnWriteArrayList listeners = new CopyOnWriteArrayList();
```

Comment:
The `CopyOnWriteArrayList` is used without specifying a generic type. This can lead to type safety issues. Consider specifying the type, e.g., `CopyOnWriteArrayList<EventListener>`.